### DISCUSSION TITLE
Arduino Due CNC Protoneer v3

### DISCUSSION THREAD

**Discussion by WZdunski** [grblHAL/core#120] [General]
Hi guys, my previous >>link>> [discussion](https://github.com/terjeio/grblHAL/discussions/350#discussion-3784527) has been closed. I don't know if this is a good place to start this topic but in other places like here (https://github.com/grblHAL/SAM3X8E) I can't start a **discussion**.

From previous discussions, I ended up having trouble connecting to the Arduino Due Atmel SAM3X8E board in LIGHTBURN.

Well, the situation is like this. After starting the LB, it can see **by Native USB** the board well, but when it needs to refresh the connection or something else will happen, showing in the console "waiting for connection". The connection cannot be restored in any other way than by closing the application LB, which is annoying because everything has to be set up again in the project to work. 

Does anyone have an idea what the problem is?
This problem is only with **LightBurn**, other programs connect every time, reset the connection, and everything works. I checked on Windows and LightBurn has the same problem with this board. Please look at my previous [discussion](https://github.com/terjeio/grblHAL/discussions/350), there are a few descriptions of this problem.

Macbook Pro - osx Monterey
I attached debug log from LightBurn 1.0.06. 
[LightBurnLog.txt](https://github.com/grblHAL/core/files/8148788/LightBurnLog.txt)

my settings at "my_machine" 
![Screenshot 2022-02-27 at 14 01 10](https://user-images.githubusercontent.com/96832902/155885504-596f7459-2f0d-4ced-9cb4-d6d033c69ac8.png)

[VER:1.1f.20220215:]
[OPT:VNMSL,35,1024,3,0]
[NEWOPT:ENUMS,RT+,TC,SED]
[FIRMWARE:grblHAL]
[NVS STORAGE:*FLASH]
[DRIVER:SAM3X8E]
[DRIVER VERSION:220111]
[BOARD:Protoneer v3]

**Comment by terjeio** [comment]
Have you tried with [compatibility level](https://github.com/grblHAL/core/wiki/Compatibility-level) set > 0?

From the log it seems a connection is established and then ignored:

```
14:11:59.543  D: O: "$X\n"
14:11:59.544  D:   I: "ok\r\n"
14:12:10.948  D: "Waiting for response..."  busy:  true  state:  5
14:12:10.949  D: O: "$I\n"
14:12:10.949  D:   I: "[VER:1.1f.20220215:]\r\n[OPT:VNMSL,35,1024,3,0]\r\n"
14:12:10.949  D:   I: "[NEWOPT:ENUMS,RT+,TC,SED]\r\n[FIRMWARE:grblHAL]\r\n[NVS STORAGE:*FLASH]\r\n[DRIVER:SAM3X8E]\r\n"
14:12:10.950  D:   I: "[DRIVER VERSION:220111]\r\n[BOARD:Protoneer v3]\r\nok\r\n"
14:12:18.279  D: Enumerating ports:
```

Or maybe not...:

```
14:13:01.051  D: Port open succeeded
14:13:01.094  D: "starting"  busy:  false  state:  0
14:13:01.094  D: EV: 1001 took 37 uS
14:13:01.095  D: EV: 1001 took 8 uS
14:13:01.095  D: EV: 1018 took 11 uS
14:13:01.095  D: EV: 1015 took 831 uS
14:13:02.183  D: "Attempting to connect"  busy:  false  state:  1
14:13:02.183  D: "Connecting..."  busy:  false  state:  1
14:13:02.185  D: O: "G0\n"
14:13:02.440  D: "Connecting..."  busy:  false  state:  1
14:13:02.443  D: O: "G0\n"
14:13:02.697  D: "Connecting..."  busy:  false  state:  1
```

Have you asked Ligthburn support about how to interpret the log? I certainly do not know how to interpret it.

**Reply by WZdunski** [reply]
I will ask on the https://forum.lightburnsoftware.com/ or maybe someone from @LightBurnSoftware will be here to help.

**Reply by WZdunski** [reply]
Hi, @terjeio @LightBurnSoftware 
After a few tests still I had a problem with the machine. I did upgrade to 1.1.0.0 new version LightBurn and that not helped. 
I made a downgrade to 1.0.0.0 and connections are stabiles. For now, I don't see problems but I need more time for tests.
